### PR TITLE
[mqtt.homeassistant] More readable label for discovered things

### PR DESCRIPTION
**Pull request by ssalonen**
Simplifies labels for discovered things, more readable and compact

Real life examples of unreadable labels before this PR

- `"RuuviTag (Sensor, Sensor, Sensor, Sensor, Sensor, Sensor, Sensor, Sensor, Sensor, Sensor)"` for https://ruuvi.com/ discovered via MQTT via [OpenMQTTGateway](https://docs.openmqttgateway.com/)
- `""OpenMQTTGateway_2 (Switch, Sensor, Switch, Switch, Sensor, Switch, Switch, Switch, Sensor, Sensor, Switch, Sensor, Switch, Sensor, Sensor, Sensor)"` for internal diagnostic sensors and switches for ESP32 running [OpenMQTTGateway](https://docs.openmqttgateway.com/)

### CODE REVIEW

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java [hunk 1]
```diff
@@ -126,6 +130,33 @@ public Set<ThingTypeUID> getSupportedThingTypes() {
...
 
+    /**
+     * Summarize components such as {Switch, Switch, Sensor} into string "Sensor, 2x Switch"
+     *
+     * @param componentNames stream of component names
+     * @return summary string of component names and their counts
+     */
+    static String getComponentNamesSummary(Stream<String> componentNames) {
+        StringBuilder summary = new StringBuilder();
+        @SuppressWarnings("null")
```
Comment (lolodomo) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java:141]: Can you please avoid that line? 
For what reason did you add it?

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java [hunk 2]
```diff
@@ -166,11 +197,23 @@ public void receivedMessage(ThingUID connectionBridge, MqttBrokerConnection conn
             thingIDPerTopic.put(topic, thingUID);
 
             // We need to keep track of already found component topics for a specific thing
-            Set<HaID> components = componentsPerThingID.computeIfAbsent(thingID, key -> ConcurrentHashMap.newKeySet());
-            components.add(haID);
+            final List<HaID> components;
+            {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java:201]: Why adding a bl{ } ?

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by lolodomo** [comment]
Looks like this PR also includes PR #13401 ? 
So, should rebased once PR #13401 is merged?

**Comment by ssalonen** [comment]
@lolodomo oh yes, indeed that is correct

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java (hunk 2 above)
Comment (ssalonen) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java:201]: Commented in the other PR

**Comment by lolodomo** [comment]
You can now rebased this PR, the other PR is merged.

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java (hunk 1 above)
Comment (ssalonen) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java:141]: The warning is 
```
Unsafe null type conversion (type annotations): The value of type 'Collector<@NonNull String,capture#of ?,Map<@NonNull String,@NonNull Long>>' is made accessible using the less-annotated type '@NonNull Collector<? super @NonNull String,Object,@NonNull Map<@NonNull String,Long>>'
```

At least one type annotation difference there seems to be `@NonNull Long` vs `Long`?

Comment (ssalonen) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java]: Resolved with trick in https://github.com/openhab/openhab-addons/pull/13402/commits/ad0a1bf1034b384bcfda003c881acfefd1b3ec1f

**Comment by ssalonen** [comment]
Rebased and code review comments addressed.

Btw, noticed that there are some SAT findings (unrelated to this PR)
```
[INFO] --- sat-plugin:0.13.0:report (sat-all) @ org.openhab.binding.mqtt.homeassistant ---
[INFO] Individual report appended to summary report.
[WARNING] Code Analysis Tool has found: 
 0 error(s)! 
 2 warning(s) 
 1 info(s)
[WARNING] OH-INF.thing.homeassistant-thing.xml:[0]
Missing the supported bridge with id broker
[WARNING] OH-INF.config.homeassistant-channel-config.xml:[0]
Unused configuration reference with uri - channel-type:mqtt:ha-channel
```

**Review by lolodomo — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
@antroids : could you please have a look?

**Comment by antroids** [comment]
> @antroids : could you please have a look?

LGTM
The labels of discovered components should more readable with this PR.

**Pull request merged**